lib/request.js

```diff
@@ -21,7 +21,18 @@ module.exports = (options) => {
     const httpModule = options.protocol === 'https:' ? https : http;
     return new Promise((resolve, reject) => {
         let hasRequestEnded = false;
+        const startTime = Date.now();
+        const timings = {};
         const request = httpModule.request(options, (response) => {
+            if (options.timing) {
+                if (timings.lookup === undefined) {
+                    timings.lookup = timings.socket;
+                }
+                if (timings.connect === undefined) {
+                    timings.connect = timings.socket;
+                }
+                timings.response = Date.now() - startTime;
```

Comment:
It's preferable to use [process.hrtime()](https://nodejs.org/api/process.html#process_process_hrtime_time) to measure intervals in node, also will give you higher-precision metrics.